planner/core/rule_build_key_info.go

```diff
@@ -235,9 +241,20 @@ func checkIndexCanBeKey(idx *model.IndexInfo, columns []*model.ColumnInfo, schem
 		}
 	}
 	if ok {
-		return newKey
+		return nil, newKey
 	}
-	return nil
+
+	// the unique index whose NotNullFlag is false since equivalence conditions can filter out null values,
+	// so a unique index without a NotNullFlag can return at most one row.
+	for _, idxCol := range idx.Columns {
+		for i, col := range columns {
+			if idxCol.Name.L == col.Name.L {
+				uniqueKey = append(uniqueKey, schema.Columns[i])
+			}
+		}
+	}
```

Comment:
Can we combine this loop with the above one?

Reply:
done